packages/react-next/src/components/Link/Link.tsx

```diff
@@ -1,40 +1,10 @@
-import * as React from 'react';
-import { getGlobalClassNames, ITheme } from '../../Styling';
-import { memoizeFunction, styled } from '../../Utilities';
-import { LinkBase } from './Link.base';
-import { ILinkProps, ILinkStyleProps, ILinkStyles } from './Link.types';
+import { compose } from '@fluentui/react-compose';
+import { LinkBase } from './LinkBase';
 import * as classes from './Link.scss';
-import { css } from 'office-ui-fabric-react/lib/Utilities';
+import { LinkProps } from './Link.types';
 
-const GlobalClassNames = {
-  root: 'ms-Link',
-};
-
-const getStaticStylesMemoized = memoizeFunction(
-  (theme: ITheme, className?: string, isButton?: boolean, isDisabled?: boolean) => {
-    const globalClassNames = getGlobalClassNames(GlobalClassNames, theme);
-
-    const propControlledClasses = [isButton && classes.button, isDisabled && classes.disabled];
-
-    const rootStaticClasses = [isDisabled && 'is-disabled'];
-
-    return {
-      root: css(className, classes.root, globalClassNames.root, ...rootStaticClasses, ...propControlledClasses),
-    };
-  },
-);
-
-const getStaticStyles = (props: ILinkStyleProps): Required<ILinkStyles> => {
-  const { className, isButton, isDisabled, theme } = props;
-
-  return getStaticStylesMemoized(theme!, className, isButton, isDisabled);
-};
-
-export const Link: React.FunctionComponent<ILinkProps> = styled<ILinkProps, ILinkStyleProps, ILinkStyles>(
-  LinkBase,
-  getStaticStyles,
-  undefined,
-  {
-    scope: 'Link',
-  },
-);
+export const Link = compose<'a', {}, {}, LinkProps, LinkProps>(LinkBase, {
+  classes,
+  slots: {},
```

Comment:
remove?

Reply:
Problem we've discussed is:

* styles prop won't work
  * `<Link styles={{ root: { background: red }}`
  * `<Link styles={ props => { root: { background: props.theme.palette.themePrimary } } }`
* Customizer injected styles wont work
  * `<Customizer scopedSettings={{ Link: { styles: { root: { ... } } }}`

For the sake of the PR, we could keep backwards compat by just using `styled` and validate the above scenarios.

Reply:
I'm using `styled` now and I added examples and visual regression tests to validate these scenarios.